src/App.js

```diff
@@ -5,7 +5,37 @@ import Slider from 'rc-slider';
 import './App.css';
 import 'rc-slider/assets/index.css';
 
+function generateInitialGridState({ width = 1, height = 1}) {
```

Comment:
This function is only used once at the mount of the App component. It just generates a model for a sequencer grid of the specified dimensions. I've modelled the grid as a 2-d matrix with the columns on the outside (and therefore referenced first when unwrapping) because I'm thinking of each column of the grid as also a step in the sequence. Later on we might want to think of each row as an instrument deck or something.